src/Sulu/Bundle/WebsiteBundle/Controller/SitemapController.php

```diff
@@ -25,59 +22,137 @@
 class SitemapController extends WebsiteController
 {
     /**
-     * Returns a rendered xmlsitemap.
+     * Render sitemap-index of all available sitemap.xml files.
+     * If only one provider exists this provider will be rendered directly.
      *
-     * @return Response
+     * @param Request $request
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
      */
     public function indexAction(Request $request)
     {
-        /** @var RequestAnalyzerInterface $requestAnalyzer */
-        $requestAnalyzer = $this->get('sulu_core.webspace.request_analyzer');
-
-        /** @var SitemapXMLGeneratorInterface $sitemapXMLGenerator */
-        $sitemapXMLGenerator = $this->get('sulu_website.sitemap_xml_generator');
-
-        $sitemap = $this->get('sulu_content.content_repository')->findAllByPortal(
-            $requestAnalyzer->getPortal()->getXDefaultLocalization()->getLocalization(),
-            $requestAnalyzer->getPortal()->getKey(),
-            MappingBuilder::create()
-                ->addProperties(['changed'])
-                ->setResolveUrl(true)
-                ->getMapping()
+        $dumpDir = $this->getDumpDir($request);
+        if ($this->get('filesystem')->exists($dumpDir . '/sitemap.xml')) {
+            return $this->setCacheLifetime(new BinaryFileResponse($dumpDir . '/sitemap.xml'));
```

Comment:
I see, so the problem is if sub folders are used as the root of webspaces? Because in a single webspace environment it would be possible.